### ISSUE TITLE
Cover integratin between search, table and root page

### ISSUE THREAD

**Issue by michal-murawski-90poe** [louisotto/github-react-repos#3]
We have some unit tests to cover the basic functionality of the view components. Now, we would like to test them all together.

Could you add an integration test (the best approach would be with @ testing-library/react), where we mock the apollo response for a specific query and assert the full async flow of a request? Again, we want to be sure that the expected user interaction works.

**Comment by louisotto** [comment]
Hi Michal,

This should be addressed in commit 1c354d064d8b4822c618c820ef3984726eddd54e

Please run `npm install` before running the project to install the new dependencies.

I've run into a bug that really frustrated me, I hope you could help me understand what I've done wrong! I get the below error when running `rpm run test`

```
Expected variables: {"search":"react language:javascript stars:&gt;10000 sort:stars"}

Failed to match 1 mock for this query. The mocked response had the following variables:
  {"search":"react language:javascript stars:&gt;10000 sort:stars"}

```

Unless I'm mistaken, I'm submitting exactly what should be expected. I've gone through docs and helpful threads like this one: https://github.com/apollographql/react-apollo/issues/617 but I can't find a solution.

Sorry to submit a failing piece of code but I'm really hoping you understand my confusion!

**Comment by michal-murawski-90poe** [comment]
Hi Louis! Thanks for the message; I will look into it today, as yesterday I had a day off.

**Comment by michal-murawski-90poe** [comment]
I had a moment to look into it. The solution to avoid this problem is to use the backslash escape character. The backslash `\` escape character turns special characters into string characters.

mocks: 
```
variables: {
        search: "react language:javascript stars\:\>10000 sort:stars",
      },
```
search
```
    variables: {
      search: `${search} language:javascript stars\:\>10000 sort:stars`,
    },
```
filter
```
          refetch({
            search: `${e.target.value} language:javascript stars\:\>10000 sort:stars`,
          });
```

**Comment by louisotto** [comment]
Hi Michal,

Apologies for the delay, I had some time off too :)

That's amazing, great catch. I didn't even consider looking at escaping because the terminal output matched, but this is something I'll remember in case it ever comes up again!

**Comment by louisotto** [comment]
Those should be fixed in 8acb04d7c0e0f1745d037e58f5fe2ba4eeb4ef4d along with a tidy up.
